[16.0][MIG] account_asset_management: Migration to 16.0

RodrigoBM opened the pull request:
https://github.com/OCA/account-financial-tools/issues/1472

rdeheele committed:
[FIX] test floats with assertAlmostEqual

[UPD] add places arg in assertAlmostEqual tests

[UPD] flake clean

lepistone committed:
move account_asset_management* out of __unported__

add asset management modules

asset mgt update

redo

synch asset mgt with recent V7 changes

lepistone committed:
make *asset* installable again

Commit:
[FIX] Inherited model is a transient model

sbidoul committed:
[IMP] account_asset_management: use reload_on_button instead of reloading page

Commit:
[IMP][account_asset_management] Define a list of move's fields that can't be modified if a move is linked with a depriciation line

[IMP][account_asset_management] Define FIELDS_AFFETCS_ASSET_MOVE as a set directly

luc-demeyer committed:
[IMP] account_asset_management: code cosmetics

luc-demeyer committed:
account_asset_management: group invoice lines fix

luc-demeyer committed:
analytic enhancement

Commit:
[ADD][account_asset_management] Add check on account move line

Commit:
[FIX][account_asset_management] Add items on affects fields list

sbidoul committed:
[FIX] account_asset_management: do not propose special and closed periods for asset posting

Commit:
[FIX][account_asset_management] Avoid to select closed or cancelled analytic account

sbidoul committed:
[UPD] prefix versions with 8.0

luc-demeyer committed:
[10.0][MIG]account_asset_management suite refactoring + 10.0 port

grindtildeath committed:
[FIX+IMP] account_asset_management: 2 things:

* Fix compute methods dependencies and small optimizations
* Rename demo file to test and move it into the right folder

luc-demeyer committed:
[FIX] account_asset_management: Test data + onchange

grindtildeath committed:
[MIG] account_asset_management: Migration to 11.0

sbidoul committed:
[IMP] asset mgt: use exclude manifest key

This is now supported natively by Odoo 11.

norlinhenrik committed:
[MIG] account_asset_management: Migration to 12.0

MaxyMoos committed:
account_asset_management: Add unit test for invoices with multiple assets

MaxyMoos committed:
Included PR #828 Fix computation of depreciation lines when having multiple assets in invoice

sbejaoui committed:
Finish 12.0 migration of account_asset_management

* account_asset: Do not loop on all the lines to search for one linked asset

Before this change, the use of `mapped` on self did loop on all the move
lines that are included in self to get the assets, what could be very
costly for a simple write on a lot of move lines. As the goal is to raise
an error only if at least one move is linked to an asset, we break the
loop if the condition is fulfilled.

* performance improvement

* [RMV] - Remove useless dependency

In 12.0 account_fiscal_year is a standard feature no need to depend on oca
module account_fiscal_year

norlinhenrik committed:
[12.0][FIX] account_asset_management: date_range -> account_fiscal_year, account_analytic_id without domain

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 40.1% (79 of 197 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

Saran440 committed:
[12.0][IMP] account_asset_management

eduaparicio committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (197 of 197 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

eduaparicio committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (195 of 195 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

svalaeys committed:
Translated using Weblate (French)

Currently translated at 100.0% (195 of 195 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/fr/

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 91.8% (179 of 195 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/es/

pedrobaeza committed:
[FIX] account_asset_management: Don't depend on account.fiscal.year created records

* There's no need of forcing to create such records if your fiscal years are regular ones.
* FY date range computation duplicates code that is already on core.
* DummyFY is a good idea, but was not used at all. Now it is.

pedrobaeza committed:
[FIX] account_asset_management: Provide hook _compute_depreciation_amount_per_fiscal_year

This way, other method_time mechanism like localization ones is able to overwrite it.

You also got an error of variable `number` not set due to previous code. Although no
alternate implementation for a new method_time, now you don't have any error,
fallbacking to standard proportional repartition.

pedrobaeza committed:
[FIX] account_asset_management: Invisible conditions must be independent from extra method_time values

If not, added methods won't hide these fields.

luc-demeyer committed:
[FIX][12.0]asset_management - fix date string compare

norlinhenrik committed:
[MIG] 11.0 account_asset to 12.0 account_asset_management

norlinhenrik committed:
[IMP] removed type, parent_id, parent_path, child_ids from account.asset & account.asset.profile

pedrobaeza committed:
[MIG+FIX+IMP] account_asset_management: Rework migration scripts + make asset group m2m

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (193 of 193 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

luc-demeyer committed:
[IMP] account_asset_management: Add reporting

Saran440 committed:
[ADD] options 'use_leap_years'

Commit:
[ADD] allows setting a depreciation ending date

Commit:
[12.0][FIX][account_asset_management] local variable 'asset' referenced before assignment

tools/account_asset_management/models/account_asset.py", line 970, in _compute_entries
    % (asset.name, asset.code) or asset.name
UnboundLocalError: local variable 'asset' referenced before assignment

pedrobaeza committed:
[MIG] account_asset_management: Handle account_asset_disposal migration

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (200 of 200 strings)

Translation: account-financial-tools-12.0/account-financial-tools-12.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-12-0/account-financial-tools-12-0-account_asset_management/pt_BR/

pedrobaeza committed:
[MIG] account_asset_management: Handle account_asset_analytic migration

Only field renaming is needed.

pedrobaeza committed:
[FIX] account_asset_management: Fix tests on leap year

It's bad idea to let dates to be variable.

luc-demeyer committed:
[12.0][FIX]traceback when deviating FY

svalaeys committed:
Fixed dynamic build of the name of the view to be called when errors are found

HviorForgeFlow committed:
[12.0] account_financial_tools: Lint fixing

ernestotejeda committed:
[IMP] account_asset_management: black, isort

ernestotejeda committed:
[MIG] account_asset_management: Migration to 13.0

pedrobaeza committed:
[FIX] account_asset_management: Avoid error

Steps to reproduce the problem:

* Go to assets view
* Group by profile
* Unfold a group and click on an asset
* Click on "Journal Entries" smart-button
* Go back to the asset list
* Click again on the same asset (or another).
* Click on "Journal Entries" smart-button

Current behavior:

Error saying "KeyError: 'profile_id'"

Expected behavior:

No error

The cause for this is that Odoo stores in the context the key `group_by` with the
value `profile_id` in the specified chain of steps. That context entry is used for
grouping records in the list, and system tries to group the journal entries also
by that field, which doesn't exists in the other model, and thus the error.

We avoided it copying the context to be passes and leaving out that entry.

pedrobaeza committed:
[IMP] account_asset_management: Simplify code for opening related entries

We don't need the search having the one2many field and no need also for list
comprehension having mapped operator.

carlosdauden committed:
[IMP] account_asset_management: Improve code and translation

Commit:
[FIX] account_asset_management: Fix tests

qgroulard committed:
[IMP] account_asset_management: Depreciation board compute upon asset confirmation

astirpe committed:
Fix of pre-commit: E741 ambiguous variable name 'l'

ValentinVinagre committed:
[IMP] account_asset_management: Add scheduled action for the depreciation asset entries creation.

ValentinVinagre committed:
[IMP] account_asset_management: Add chatter entry in invoice for generated asset

ValentinVinagre committed:
[FIX] account_asset_management: FIX empty asset reference

If the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

Commit:
[FIX] account_asset_management: fix remaining_value cache miss error.

HaraldPanten committed:
Translated using Weblate (Spanish)

Currently translated at 87.7% (171 of 195 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/es/

FernandoRomera committed:
[FIX] account_asset_management:
* multicompany error

FernandoRomera committed:
[FIX] version

[FIX] new line

[FIX] new line

[FIX] without openupgrade

[FIX] without openupgrade

[FIX] without openupgrade

[FIX] without openupgrade

[FIX] without openupgrade

Saran440 committed:
[FIX] account_asset_management - Allow editing salvage value

pedrobaeza committed:
[IMP] account_asset_management: Refinements

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 39.9% (77 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 39.9% (77 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 40.4% (78 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 42.0% (81 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 48.2% (93 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 49.7% (96 of 193 strings)

Translation: account-financial-tools-13.0/account-financial-tools-13.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-13-0/account-financial-tools-13-0-account_asset_management/zh_CN/

Saran440 committed:
[FIX] calculate days when remove asset

fmdl committed:
[FIX] account_asset_management : empty message is posted if there are no asset

pedrobaeza committed:
[FIX] account_asset_management: Proper asset unlink + move unlink

2 related fixes:

- When removing an asset line depreciation move, we have to pass it
  first to draft, or we won't be able to remove it even with the
  context.
- When removing a move, the check for removing the linked asset should
  be only for purchase documents, not for "not sale" documents.

joao-p-marques committed:
[MIG] account_asset_management: Migration to v14

TT26410

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 59.0% (114 of 193 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/nl/

pedrobaeza committed:
[FIX] account_asset_management: Missing field renaming

type > move_type, due to uncovered code.

StefanRijnhart committed:
[FIX] account_asset_management: don't set class attributes in reporting

Also, actually generate report in test plus other tests to make it to the
codecov target

JordiBForgeFlow committed:
[13.0][fix][account_asset_management] display only accounts associated
to the current company.

JordiBForgeFlow committed:
[13.0][account_asset_management] button to navigate from the vendor bill
to the asset.

JordiBForgeFlow committed:
[13.0][account_asset_management] propose account when user chooses

MateuGForgeFlow committed:
[13.0][account_asset_management] propose account when user chooses

victoralmau committed:
[FIX] account_asset_management: Fix onchange account_id and asset_profile_id to prevent account_id is empty in some use cases

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_asset_management 14.0.1.0.2

victoralmau committed:
[FIX] account_asset_management: Prevent create lines with init=True when account_lock_date_update addon is installed and lock date defined

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 95.8% (185 of 193 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/fr/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 98.9% (191 of 193 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/fr/

yelizariev committed:
[FIX] account_asset_management: correct field name

OCA-git-bot committed:
account_asset_management 14.0.1.0.3

joao-p-marques committed:
[IMP] account_asset_management: Add analytic tags and propagate

TT28974

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 97.9% (189 of 193 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/fr_FR/

JordiMForgeFlow committed:
account_asset_management: fix group_assets sorting

JordiMForgeFlow committed:
[13.0][account_asset_management]: add user error when creating asset from bill

StefanRijnhart committed:
[FIX] account_asset_mamagement: set invoice date

Adapt to https://github.com/odoo/odoo/commit/fcaa54939e, which prevents
setting a default invoice date for purchase documents

OCA-git-bot committed:
account_asset_management 14.0.1.0.4

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 14.0.1.0.5

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 66.3% (128 of 193 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/nl/

MiquelRForgeFlow committed:
[FIX] account_asset_management: include 'number' method_time

MiquelRForgeFlow committed:
[FIX] account_asset_management: create asset using Form

This way, the computes are computed as expected.

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_asset_management 14.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

victoralmau committed:
[FIX] account_asset_management: Fix error from group_ids in asset when set profile_id

victoralmau committed:
[FIX] account_asset_management: Fix error according to _name_search in group when try to set groups in profile

OCA-git-bot committed:
account_asset_management 14.0.1.1.1

JordiMForgeFlow committed:
[13.0][FIX]account_asset_management: use column_invisible for fields inside a tree of a o2m field

JordiMForgeFlow committed:
[13.0][IMP]account_asset_management: add optional attribute for added columns in invoice lines

OCA-git-bot committed:
account_asset_management 14.0.1.2.0

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 14.0.2.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 99.0% (218 of 220 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/fr/

JordiBForgeFlow committed:
[IMP] account_asset_management: add carry_forward_missed_depreciations

JordiBForgeFlow committed:
[IMP] account_asset_management: Allow to reverse the posting of a depreciation line
instead of deleting the journal entry.
This will be done when the company has activated the inalterability hash
on the original journal entry.

OCA-git-bot committed:
account_asset_management 14.0.2.0.1

ps-tubtim committed:
[14.0][FIX] account_asset_management: pre-commit

pedrobaeza committed:
[FIX] account_asset_management: Don't put asset name as entry name

This can provoke an error due to duplicated entry name.

Remaining of the migration to version 13.0, and similar to
https://github.com/OCA/account-financial-tools/commit/b7637cbe45701700148efd6d9e32fb0c0079b4f2#diff-5318cd3560cdb2d23d2889fe39c5b71745c140810b20f64f0c9aa5b24b8b602bL206

Commit:
[FIX] account_asset_management: make asset company match move's

To start, the context `company_id` is no longer used in v14. Instead, we use `with_company`.

OTOH, it doesn't make much sense because in case you have several companies enabled in your context, not necessarily your main one is gonna be the one that should be used for the asset. Instead, it should be the move company always, or consistency rules will fail.

Altered forward port of https://github.com/OCA/account-financial-tools/pull/1223.

@Tecnativa TT31311

OCA-git-bot committed:
account_asset_management 14.0.2.0.3

ps-tubtim committed:
[FIX] account_asset_management: fix _check_dates

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_asset_management 14.0.2.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

kittiu committed:
[14.0][FIX] account_asset_management, error on remove for asset no depre

kittiu committed:
[14.0][FIX] account_asset_management, fix multi currency

ps-tubtim committed:
[ENH] account_asset_management: Add purchase value in asset report

MiquelRForgeFlow committed:
[IMP] account_asset_management: hook for asset vals

OCA-git-bot committed:
account_asset_management 14.0.2.2.0

OCA-git-bot committed:
account_asset_management 14.0.2.2.1

kittiu committed:
[14.0][FIX] account_asset_management, singleton error on asset_count

OCA-git-bot committed:
account_asset_management 14.0.2.2.2

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 14.0.2.2.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

kittiu committed:
[14.0][FIX] account_asset_management, error create bill by purchase user

OCA-git-bot committed:
account_asset_management 14.0.2.3.0

alexis-via committed:
account_asset_management: improve and cleanup views

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 14.0.2.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

victoralmau committed:
[FIX] account_asset_management: Avoid the error when clicking on "Delete" button if there is a residual value and all the depreciation lines are posted.
TT32861

ps-tubtim committed:
[FIX] account_asset_management: sudo smart button asset

OCA-git-bot committed:
account_asset_management 14.0.2.4.1

RachidAlassir committed:
Translated using Weblate (Arabic)

Currently translated at 34.5% (75 of 217 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/ar/

etobella committed:
[UPD] account_asset_management: Mark as mature

OCA-git-bot committed:
[UPD] README.rst

sbidoul committed:
Bump account_asset_management version

pedrobaeza committed:
account_asset_management 14.0.2.4.3

JordiBForgeFlow committed:
[IMP] account_asset_management: assets can be removed irrespective of the Time Method used
It does not make sense that you cannot remove an asset where the time method used
is not based on number of years or end date. If you choose, for example, to depreciate the asset
using the time method 'Number of depreciations' you will want to remove the asset when it has been
fully depreciated.

OCA-git-bot committed:
account_asset_management 14.0.2.4.4

Saran440 committed:
[FIX] calculate day mistake, remove asset

mariadforgeflow committed:
[IMP] account_asset_management: set asset_profile_id on invoice creation

victoralmau committed:
[IMP] account_asset_management: Add chatter + activities to assets.
TT34459

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
account_asset_management 14.0.2.5.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

OCA-git-bot committed:
account_asset_management 14.0.2.5.1

oca-travis committed:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
account_asset_management 14.0.2.6.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/

OCA-git-bot committed:
account_asset_management 14.0.2.6.1

OCA-git-bot committed:
[IMP] update dotfiles

mariadforgeflow committed:
[IMP] account_asset_management: prevent from creating assets from tax lines

OCA-git-bot committed:
account_asset_management 14.0.2.7.0

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 26.9% (71 of 263 strings)

Translation: account-financial-tools-14.0/account-financial-tools-14.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-14-0/account-financial-tools-14-0-account_asset_management/it/

manuelregidor committed:
[IMP] : black, isort, prettier

manuelregidor committed:
[MIG]account_asset_management: Migration to 15.0

Commit:
[UPD] Update account_asset_management.pot

OCA-git-bot committed:
[UPD] README.rst

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (263 of 263 strings)

Translation: account-financial-tools-15.0/account-financial-tools-15.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-15-0/account-financial-tools-15-0-account_asset_management/es_AR/

Commit:
[15.0][FIX] account_asset_management: add post-install tagged.

OCA-git-bot committed:
account_asset_management 15.0.1.0.1

LudLaf committed:
Translated using Weblate (Spanish)

Currently translated at 98.8% (260 of 263 strings)

Translation: account-financial-tools-15.0/account-financial-tools-15.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-15-0/account-financial-tools-15-0-account_asset_management/es/

LudLaf committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (263 of 263 strings)

Translation: account-financial-tools-15.0/account-financial-tools-15.0-account_asset_management
Translate-URL: https://translation.odoo-community.org/projects/account-financial-tools-15-0/account-financial-tools-15-0-account_asset_management/es/

Commit:
[MIG] account_asset_management_menu: Migration to 15.0

OCA-git-bot committed:
[UPD] README.rst

RodrigoBM commented:
pre-commit fails for double duplicate field, but I need this one in V16 for groups="base.group_multi_company" or I haven't migrated the view properly.

RodrigoBM commented:
/ocabot migration account_asset_management

pedrobaeza commented:
/ocabot migration account_asset_management

pedrobaeza commented on account_asset_management/views/account_asset_group.xml:
```diff
@@ -0,0 +1,57 @@
+            <form string="Asset Group">
+                <group>
+                    <group>
+                        <field name="company_id" invisible="1" />
```
Remove this field

RodrigoBM replied:
if I delete this, I get this error:

```
odoo.tools.convert.ParseError: while parsing /mnt/environment/16.0-mig-account_asset_management/account_asset_management/views/account_asset_group.xml:24
Error while validating view near:

<tree editable="bottom" __validate__="1">
                <field name="name"/>
                <field name="code" optional="show"/>

Field 'company_id' used in domain of field 'parent_id' ([('company_id', 'in', [company_id, False])]) is restricted to the group(s) base.group_multi_company.

View error context:
{'file': '/mnt/environment/16.0-mig-account_asset_management/account_asset_management/views/account_asset_group.xml',
 'line': 1,
 'name': 'account.asset.group.tree',
 'view': ir.ui.view(966,),
 'view.model': 'account.asset.group',
 'view.parent': ir.ui.view(),
 'xmlid': 'account_asset_group_view_tree'}
```

I have assumed that this was correct because of this example: 
https://github.com/OCA/OCB/blob/16.0/addons/account/views/account_account_views.xml#L92

RodrigoBM replied:
The problem is in the parent_id field.
```
parent_id = fields.Many2one(
        comodel_name="account.asset.group",
        string="Parent Asset Group",
        ondelete="restrict",
        check_company=True,
    )
```
If the company_id field with group constraint "base.group_multi_company" is set in view. This parent_id field will not pass the restrictions added here.

https://github.com/OCA/OCB/blob/16.0/odoo/addons/base/models/ir_ui_view.py#L2974

This is why you need the duplicate field with the invisible property.

pedrobaeza replied:
Yeah, it makes sense. @moylop260 it seems the linter shouldn't rise on this case. I think the duplicated field in view should take into account `groups` attribute.

moylop260 replied:
I have checked the lint in odoo project itself and it is detecting the following cases: 

```txt
odoo/addons/account/views/account_move_views.xml:166 Duplicate xml field "company_id" in lines 167
odoo/addons/account/views/account_account_views.xml:92 Duplicate xml field "company_id" in lines 104
odoo/addons/account/views/account_journal_views.xml:253 Duplicate xml field "company_id" in lines 257
odoo/addons/hr/views/hr_employee_views.xml:276 Duplicate xml field "company_id" in lines 281
odoo/addons/hr_holidays/views/hr_leave_stress_day_views.xml:32 Duplicate xml field "company_id" in lines 34
odoo/addons/hr_recruitment/views/hr_recruitment_views.xml:52 Duplicate xml field "company_id" in lines 53
odoo/addons/product/views/product_pricelist_item_views.xml:60 Duplicate xml field "company_id" in lines 85
odoo/addons/project/views/project_views.xml:691 Duplicate xml field "company_id" in lines 692
odoo/addons/project/views/project_views.xml:1509 Duplicate xml field "company_id" in lines 1510
odoo/addons/resource/views/resource_views.xml:61 Duplicate xml field "company_id" in lines 64
odoo/addons/stock/views/product_strategy_views.xml:8 Duplicate xml field "company_id" in lines 37
odoo/addons/stock/views/stock_quant_views.xml:111 Duplicate xml field "company_id" in lines 120
odoo/addons/stock/views/stock_quant_views.xml:363 Duplicate xml field "company_id" in lines 387
odoo/addons/stock/views/stock_orderpoint_views.xml:43 Duplicate xml field "company_id" in lines 62
odoo/addons/stock_picking_batch/views/stock_picking_batch_views.xml:63 Duplicate xml field "company_id" in lines 76
```

So, I think it is a valid case

However, we are migrating this [pylint-odoo](https://github.com/OCA/pylint-odoo) check to [new project](https://github.com/OCA/odoo-pre-commit-hooks) and it will be deprecated in pylint itself and we are working in progress in the new project


In order to avoid delaying your contribution you even can disable this check [only for this file](https://github.com/oca/pylint-odoo#skip-one-xml-check) or you can [remove the check for this project](https://github.com/OCA/account-financial-tools/blob/f9ddfe348fa7cd54e2a7e10482b69d1da65a7164/.pylintrc-mandatory#L28)

In the next sync of configuration files I hope fix this issue in the new tool

moylop260 replied:
FYI https://github.com/OCA/odoo-pre-commit-hooks/issues/10

pedrobaeza replied:
Thanks for checking. Yeah, disabling it for this file can work for now. @RodrigoBM can you do it?

RodrigoBM replied:
Yeah, I do it

pedrobaeza commented on account_asset_management/models/account_asset.py:
```diff
@@ -275,13 +275,6 @@ class AccountAsset(models.Model):
         readonly=False,
         store=True,
     )
-    analytic_tag_ids = fields.Many2many(
```
Please replace it by the corresponding new feature

RodrigoBM replied:
i think its ok now

pedrobaeza commented on account_asset_management/models/account_asset.py:
```diff
@@ -441,9 +429,9 @@ def _onchange_purchase_salvage_value(self):
-        asset = super().create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        asset = super().create(vals_list)
```
This returns several records, so you have to take it into account.

RodrigoBM replied:
i think its ok now

pedrobaeza commented on account_asset_management/models/account_asset_line.py:
```diff
@@ -248,11 +246,12 @@ def _setup_move_line_data(self, depreciation_date, account, ml_type, move):
             "asset_id": asset.id,
         }
+        if analytic_id:
+            move_line_data["analytic_distribution"][analytic_id] = 100
```
This is not the same thing as the previous tags.

RodrigoBM replied:
i think its ok now

pedrobaeza commented on account_asset_management/models/account_asset_profile.py:
```diff
@@ -16,9 +16,6 @@ class AccountAssetProfile(models.Model):
     account_analytic_id = fields.Many2one(
         comodel_name="account.analytic.account", string="Analytic account"
     )
-    analytic_tag_ids = fields.Many2many(
```
Replace it by the corresponding feature

RodrigoBM replied:
done

pedrobaeza commented on account_asset_management/models/account_asset_profile.py:
```diff
@@ -207,11 +204,12 @@ def _compute_prorrata(self):
+        for vals in vals_list:
+            if vals.get("method_time") != "year" and not vals.get("prorata"):
+                vals["prorata"] = True
+        profile = super().create(vals_list)
```
This returns several records. Take it into account.

RodrigoBM replied:
i think its ok now

pedrobaeza commented on account_asset_management/models/account_move.py:
```diff
@@ -81,7 +81,6 @@ def _prepare_asset_vals(self, aml):
             "purchase_value": depreciation_base,
             "partner_id": aml.partner_id,
             "date_start": self.date,
-            "account_analytic_id": aml.analytic_account_id,
```
Replace it by the corresponding feature.

RodrigoBM replied:
done

pedrobaeza commented on account_asset_management/report/account_asset_report_xls.py:
```diff
@@ -23,11 +22,6 @@ class AssetReportXlsx(models.AbstractModel):
     _description = "Dynamic XLS asset report generator"
     _inherit = "report.report_xlsx.abstract"
 
-    def _(self, src):
```
Why removing this?

RodrigoBM replied:
The purpose of this method is to get the translation of self.().

The def translate method does not exist in the tools.

Also, in v15 when you run the "def translate" method you also it run the query. 

	`cr.execute('select value from ir_translation where lang=%s and type=%s and 	name=%s and src=%s and md5(src)=md5(%s)', (lang, source_type, str(name), 	source, source))`

But if it is called like this:

	 `val = translate(self.env.cr, IR_TRANSLATION_NAME, "report", lang, src) or src`

The value of type is report
	
        `”type=”report”`

But type in the class ir_translation only can have this values: 
	```
TRANSLATION_TYPE = [
	('model', 'Model Field'),
	('model_terms', 'Structured Model Field'),
	('code', 'Code'),
	]
        ```

If the query does not find anything, it will always return src. 

Also in v15 is not working.

![Selección_063](https://user-images.githubusercontent.com/6359121/196727359-9fbe2d82-2f16-4e6c-b265-7832bb27ffb0.jpg)

I change it to standard Odoo and now is working.

![Selección_064](https://user-images.githubusercontent.com/6359121/196727686-88e62e0a-43ff-47f0-9d66-8ee9aa866d89.jpg)

pedrobaeza replied:
OK, the idea behind this is that the translation works having a context in the class. This trick was needed for v13 if I don't remember bad, but people doing later translations didn't check...

pedrobaeza commented on account_asset_management/tests/test_account_asset_management.py:
```diff
@@ -249,33 +242,23 @@ def test_01_nonprorata_basic(self):
         self.assertEqual(ict0.value_residual, 1000)
         vehicle0.validate()
-        created_move_ids = vehicle0.depreciation_line_ids[1].create_move()
-        for move_id in created_move_ids:
```
Keep these asserts

RodrigoBM replied:
done

pedrobaeza commented on account_asset_management/views/account_move.xml:
```diff
@@ -4,7 +4,6 @@
         <field name="name">account.move.form.account.asset.management</field>
         <field name="model">account.move</field>
         <field name="inherit_id" ref="account.view_move_form" />
-        <field name="groups_id" eval="[(4, ref('account.group_account_invoice'))]" />
```
The rest of the fields should have the group as well.

RodrigoBM replied:
i think its ok now

RodrigoBM replied:
done

pedrobaeza reviewed (changes requested):
All the analytic stuff is not correctly ported.

KNVx committed:
[FIX] account_asset_management: Poor performance updating assets massively

When updating assets massively the performance when searching for deprecation_lines_ids is very low.
It is a difficult problem to detect since massive asset updates are not common.
Indexing in this field dramatically improves the timing performance.
For example, to update 3000 records it took 30 sec. With this index it takes to 2 seconds.

alexis-via committed:
[IMP] : pre-commit stuff

FernandoRomera reviewed (approved):
Functional test with runboat, Ok.

fcvalgar reviewed (approved):
Functional review is ok. 

Review Asset Profile V16 and this model work fine.

FernandoRomera commented:
@pedrobaeza 
Can you merge this PR, please?

rafaelbn commented:
Hello @pedrobaeza , I think you comments have been attended. Thanks!

hieulucky111 commented on account_asset_management/models/account_asset_line.py:
```diff
@@ -0,0 +1,331 @@
+    depreciation_base = fields.Float(
+        related="asset_id.depreciation_base", string="Depreciation Base", readonly=True
+    )
+    amount = fields.Float(digits="Account", required=True)
```
i think we can not require Float field.

RodrigoBM replied:
I think you are right. But this is being addressed here https://github.com/OCA/account-financial-tools/pull/1513 . I think it is better to consider this as a migration issue. And add the PR change to v15 and v16 in another PR.

RodrigoBM replied:
hi i add pr https://github.com/OCA/account-financial-tools/pull/1513 and now it is ok

hieulucky111 commented on account_asset_management/models/account_asset_line.py:
```diff
@@ -0,0 +1,331 @@
+        "for which Odoo has not generated accounting entries.",
+    )
+    company_id = fields.Many2one(
+        "res.company",
```
no need to re-define model for related field

RodrigoBM replied:
You mean not adding "res.company" in the field definition? however, this is also contemplated in pr #1513.

RodrigoBM replied:
hi i add pr https://github.com/OCA/account-financial-tools/pull/1513 and now it is ok

RodrigoBM committed:
[MIG] account_asset_management: Migration to 16.0

RodrigoBM committed:
[REF] account_asset_management: the name of the company_currency_id field is changed to currency_id and currency_field is not required.

RodrigoBM commented:
Hi  @fcvalgar @FernandoRomera 

with the comment of @hieulucky111 i think is a good moment to add pr #1513 in this migration, is possible that your functional test again?

@pedrobaeza i add ref commit for change company_currency_id to currency_id and not to add currency_field option in monetary field.

Best regards

pedrobaeza commented on account_asset_management/models/account_asset_line.py:
```diff
@@ -35,9 +35,7 @@ class AccountAssetLine(models.Model):
-    amount = fields.Monetary(
-        string="Amount", required=True, currency_field="company_currency_id"
-    )
+    amount = fields.Monetary(required=True, currency_field="company_currency_id")
```
Please change also the currency on this model to `currency_id` and remove all the `currency_field` references. In fact, the related field should be to the asset currency, not to the company one (although right now is the same).

RodrigoBM replied:
I did that in my commit [[REF] account_asset_management: the name of the company_currency_id](https://github.com/OCA/account-financial-tools/pull/1479/commits/73b3200189d15bcdaa97e81def78642b42a671d6). didn't know if I should apply fixup with commit MIG.

pedrobaeza reviewed (changes requested):
If you enter to the asset form without the analytic group, there's this error:

```
You are not allowed to access 'Analytic Plan's Applicabilities' (account.analytic.applicability) records.

This operation is allowed for the following groups:
	- Technical/Analytic Accounting

Contact your administrator to request access if necessary.
```

Please check the groups associated to the analytic fields.

RodrigoBM commented:
> If you enter to the asset form without the analytic group, there's this error:
> 
> ```
> You are not allowed to access 'Analytic Plan's Applicabilities' (account.analytic.applicability) records.
> 
> This operation is allowed for the following groups:
> 	- Technical/Analytic Accounting
> 
> Contact your administrator to request access if necessary.
> ```
> 
> Please check the groups associated to the analytic fields.

I added group_analytic_accounting to analytic_distribution field.

Thanks

hieulucky111 reviewed (approved):
LGTM.

fcvalgar commented:
Now I detect an error when I apply an asset profile on the invoice line @RodrigoBM 

https://www.loom.com/share/c172198b9acd429da5e35f5400513a9f

@moduon MT-1849

RodrigoBM commented:
> Now I detect an error when I apply an asset profile on the invoice line @RodrigoBM
> 
> https://www.loom.com/share/c172198b9acd429da5e35f5400513a9f
> 
> @moduon MT-1849

I have already solved the bug. Thank you @fcvalgar

pedrobaeza reviewed (approved):
Thanks, Rodrigo!

/ocabot merge nobump
